flutter_modular/lib/src/presenters/navigation/modular_route_information_parser.dart

```diff
@@ -191,17 +191,37 @@ class ModularRouteInformationParser extends RouteInformationParser<ModularRoute>
     if (path.isEmpty) {
       throw Exception("Router can not be empty");
     }
+    var uri = Uri.tryParse(path);
+    Map<String, List<String>>? queryParams;
+    String? fragment;
+    // the fragments and query params are removed from the path so that it doesn't affect route matching etc
+    //i.e /dashboard should also match /dashboard?id=12
+    if (uri != null) {
+      queryParams = uri.queryParametersAll;
```

Comment:
I wonder if we could return just one object. However I have been looking at the URI documentation and it is a Global standard. I am not very familiar with queryParameters.